Add docs for routing to Kubernetes and Consul destinations

marcogschmidt opened the pull request:
Updated the "Advanced Route Actions" page with dedicated sections for the 3 routing destination types:
- `upstream`
- `kube`
- `consul`

scranton commented on gloo/docs/user_guides/gateway/configuring_route_options/advanced_routing_action.md:
```diff
@@ -128,7 +136,7 @@ Second, you need to add a [`subset`]({{% ref "/v1/github.com/solo-io/gloo/projec
 of the Route Action.
 
-Following is an example of using a label, e.g., `color: blue`, to subset pods handling requests.
+Following is an example of using a label, e.g. `color: blue`, to subset pods handling requests.
```
When reading this I was confused when it jumped to an Upstream manifest from a doc page on how to write Virtual Service manifests. I feed it would help if the lead into this example stated something like, "The Upstream allows you to subset all requests to that upstream as you see in the following example. Within the routeAction you can also define a similar subset that is applied only for a given route".

It feels like a little more is needed to help the user understand that they are still in the Virtual Service Routing documentation and you're referencing an Upstream as an example

marcogschmidt replied:
Done

marcogschmidt replied:
I agree, the subset configuration is not easy to understand, particularly why you need to add config in two places. I will add some more detail and a link to the [Envoy docs](https://github.com/envoyproxy/envoy/blob/master/source/docs/subset_load_balancer.md) that explain how subset load balancing works.

scranton commented on gloo/docs/user_guides/gateway/configuring_route_options/advanced_routing_action.md:
```diff
@@ -195,11 +203,134 @@ apiVersion: gateway.solo.io/v1
+If you are running Gloo in a Kubernetes cluster, it is possible to directly specify 
+[Kubernetes Services](https://kubernetes.io/docs/concepts/services-networking/service/) as routing destinations. 
+The `kube` destination type has two required fields: `ref` is a [ResourceRef]({{% ref "/v1/github.com/solo-io/solo-kit/api/v1/ref.proto.sk#resourceref" %}}) 
+to the service that should receive traffic and `port` is the port on which the service is listening.
```
I assume the port is a number field, and that the port must be one of the existing defined in the service's associated k8s Service manifest. Helpful to have a little more detail on any constraints.

marcogschmidt replied:
Updated

scranton commented on gloo/docs/user_guides/gateway/configuring_route_options/advanced_routing_action.md:
```diff
@@ -195,11 +203,134 @@ apiVersion: gateway.solo.io/v1
+#### Consul destination {#dest-consul}
+
+Gloo is capable of discovering services registered with [HashiCorp Consul](https://www.hashicorp.com/products/consul/). 
+If this feature has been enabled via the [correspondent field]({{% ref "/v1/github.com/solo-io/gloo/projects/gloo/api/v1/settings.proto.sk#servicediscoveryoptions" %}}) 
```
I'd suggest linking to the ConsulConfiguration (http://localhost:1313/v1/github.com/solo-io/gloo/projects/gloo/api/v1/settings.proto.sk/#consulconfiguration) and calling out or separately linking to the service discovery option. I clicked through the link and I landed in the middle of a table of settings and I had no idea what I was supposed to be looking at. I would not assume the reader understands about Gloo upstream discovery.

marcogschmidt replied:
Done

scranton commented on gloo/docs/user_guides/gateway/configuring_route_options/advanced_routing_action.md:
```diff
@@ -195,11 +203,134 @@ apiVersion: gateway.solo.io/v1
+        - secondary
+{{< /highlight >}}
+
+while this one will forward the same requests only to the first two instances (the ones in data center `dc1`)
```
change to 'while this next example will forward'. hard to tell if 'one' is referring to next or previous example

marcogschmidt replied:
Updated

scranton commented:
Good update. Minor tweaks requested.

marcogschmidt commented:
@scranton Thanks for the feedback, I addressed your comments. In particular, I added a more thorough explanation of the subsets. Let me know what you think.
